src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from copy import deepcopy
+from argparse import Namespace
+import numpy as np
+import torch
+
+from nni.compression.torch.compressor import Pruner
+from .channel_pruning_env import ChannelPruningEnv
+from .lib.agent import DDPG
+from .lib.utils import get_output_folder
+
+torch.backends.cudnn.deterministic = True
+
+class AMCPruner(Pruner):
+    """
+    A pytorch implementation of AMC: AutoML for Model Compression and Acceleration on Mobile Devices.
+    (https://arxiv.org/pdf/1802.03494.pdf)
+
+    Parameters:
+        model: nn.Module
+            The model to be pruned.
+        config_list: list
+            Configuration list to configure layer pruning.
+            Supported keys:
+            - op_types: operation type to be pruned
+            - op_names: operation name to be pruned
+        evaluator: function
+            function to evaluate the pruned model.
+            The prototype of the function:
+            >>> def evaluator(val_loader, model):
+            >>>     ...
+            >>>     return acc
+        val_loader: torch.utils.data.DataLoader
+            Data loader of validation dataset.
+        suffix: str
+            suffix to help you remember what experiment you ran. Default: None.
+        job: str
+            train_export: search best pruned model and export after search.
+            export_only: export a searched model, searched_model_path and export_path must be specified.
+        searched_model_path: str
+            when job == export_only, use searched_model_path to specify the path of the searched model.
+        export_path: str
+            path for exporting models
+
+        # parameters for pruning environment
+        model_type: str
+            model type to prune, currently 'mobilenet' and 'mobilenetv2' are supported. Default: mobilenet
+        flops_ratio: float
+            preserve flops ratio. Default: 0.5
+        lbound: float
+            minimum weight preserve ratio for each layer. Default: 0.2
+        rbound: float
+            maximum weight preserve ratio for each layer. Default: 1.0
+        reward: function
+            reward function type
```

Comment:
what's the choices of reward function  type?

Reply:
Thanks, doc added.